credentials/apps/badges/services/processing.py

```diff
@@ -49,6 +50,7 @@ def process_event(sender, **kwargs):
     #   - BADGE_PROGRESS_INCOMPLETE emitted >> handle_badge_regression (possibly, we need a flag here)
 
     user_data = get_user_data(kwargs)
+    event_user = identify_event_user(user_data)
```

Comment:
@andrii-hantkovskyi pls try to reuse: https://github.com/raccoongang/credentials/blob/95747fd3d628ddf22eb68f7f01c6b24b7544fac9/credentials/apps/core/api.py#L33